test/e2e-tests.sh

```diff
@@ -23,136 +23,163 @@ source $(dirname $0)/e2e-common.sh
+    echo "Here's the failed pod info"
     kubectl get pod --namespace tekton-pipelines -l app=go-hello-world -o name --namespace tekton-pipelines -o yaml
     kubectl describe pod --namespace tekton-pipelines -l app=go-hello-world --namespace tekton-pipelines
-    kubectl get pods -n tekton-pipelines
-    kubectl get deployments -n tekton-pipelines
-    echo "Dashboard logs"
-    kubectl logs -l app=tekton-dashboard -n tekton-pipelines
-    echo "Pods in all namespaces"
-    kubectl get pods --all-namespaces
-    echo "Pipelineruns as json, all namespaces"
-    kubectl get pipelineruns -o json --all-namespaces
-    echo "TaskRun info"
-    kubectl -n tekton-pipelines get TaskRun -o json
-    echo "PipelineRun info"
-    kubectl -n tekton-pipelines get PipelineRun -o json
-    echo "PipelineRun container info"
-    kubectl -n tekton-pipelines logs -l app=e2e-pipelinerun --all-containers
-    sleep 5
+    fail_test "Test Failure, go-hello-world deployment is not running, see above for the PV and pod information"
   fi
-done
-
-if [ "$deploymentExist" = "false" ]; then
-  echo "Here's the failed pod info"
-  kubectl get pod --namespace tekton-pipelines -l app=go-hello-world -o name --namespace tekton-pipelines -o yaml
-  kubectl describe pod --namespace tekton-pipelines -l app=go-hello-world --namespace tekton-pipelines
-  fail_test "Test Failure, go-hello-world deployment is not running, see above for the PV and pod information"
-fi
-
-# Ping deployment
-kubectl port-forward $(kubectl get pod  --namespace tekton-pipelines -l app=go-hello-world -o name) --namespace tekton-pipelines 8080 &
-podForwardPID=$!
-
-podCurled=false
-for i in {1..20}
-do
-  resp=$(curl -k  http://127.0.0.1:8080)
-  if [ "$resp" != "" ]; then
-    echo "Response from pod is: $resp"
-    podCurled=true
-    if [[ "$resp" = *${EXPECTED_RETURN_VALUE}* ]]; then
-      echo "PipelineRun successfully executed"
-      break
+
+  # Ping deployment
+  kubectl port-forward $(kubectl get pod  --namespace tekton-pipelines -l app=go-hello-world -o name) --namespace tekton-pipelines 8080 &
+  podForwardPID=$!
+
+  podCurled=false
+  for i in {1..20}
+  do
+    resp=$(curl -k  http://127.0.0.1:8080)
+    if [ "$resp" != "" ]; then
+      echo "Response from pod is: $resp"
+      podCurled=true
+      if [[ "$resp" = *${EXPECTED_RETURN_VALUE}* ]]; then
+        echo "PipelineRun successfully executed"
+        break
+      else
+        fail_test "PipelineRun error, returned an incorrect message: $resp"
+      fi
     else
-      fail_test "PipelineRun error, returned an incorrect message: $resp"
+      sleep 5
     fi
-  else
-    sleep 5
+  done
+
+  if [ "$podCurled" = "false" ]; then
+    fail_test "Test Failure, Not able to curl the pod"
   fi
-done
 
-if [ "$podCurled" = "false" ]; then
-  fail_test "Test Failure, Not able to curl the pod"
-fi
+  kill -9 $dashboardForwardPID
+  kill -9 $podForwardPID
+
+  uninstall_dashboard_backend $overlay
+  delete_pipeline_crd
```

Comment:
Do we need to install / uninstall pipelines for each dashboard overlay being tested or can we just do this once up front?

Reply:
As we put everything in the same namespace it helps making sure is cleaned up before moving to the next tests.

It should not be long anyway.